SNAP-735 Wix Velo plugin update

adagiomedia opened the pull request:
Fixed bugs in ex-BTCPay code, added order ID, referral code etc.

nemanja-z reviewed (changes requested):
I noticed a couple of changes that I’d like to understand better:

Function Rename: I see that you renamed the function post_coinsnap_webhook to use_post_coinsnap_webhook. Could you share the reasoning behind this rename?

Commented Conditional: The conditional block at line 14 in src/backend/https-functions.js has been commented out. Could you explain why this was done?

Aside from adding the logo and the changes mentioned above, I don’t see any other modifications in the code.

adagiomedia commented:
Nemanja, about changes:

_Function Rename: I see that you renamed the function post_coinsnap_webhook to use_post_coinsnap_webhook. Could you share the reasoning behind this rename?_

Wix didn't see function with name "post_coinsnap_webhook" that we tried to call from Coinsnap. By Wix rules it was needed to add "use_" in beginning, after that error "Function doesn't exist" has gone.

_Commented Conditional: The conditional block at line 14 in src/backend/https-functions.js has been commented out. Could you explain why this was done?_

The reason: our Wix didn't see functions startsWith() and endWith(), I have commented them because we don't need them in all the cases that BTCPay used them. 

_Aside from adding the logo and the changes mentioned above, I don’t see any other modifications in the code._

There were some modifications: 

- I have changed headers array members, 
- I have fixed path to Coinsnap App in invoice creation process, 
- After that I have changed JSON object with invoice body because we have different invoice request object structure?
- And I have added our special data there,
- I have inserted "export const refundTransaction" - plugin cannot work without it.

I have also spent some time trying to submit order number, but it is possible to send order ID only. But ... maybe I am not right.
